eng/tools/rush-runner/helpers.js

```diff
@@ -0,0 +1,149 @@
+ * @param {string[]} actionComponents - An array of strings containing the packages names to run the action on.
+ */
+export const getDirectionMappedPackages = (packageNames, actionComponents) => {
+  const mappedPackages = [];
+
+  for (const packageName of packageNames) {
+    // Build command without any additional option should build the project and downstream
+    // If service is configured to run only a set of downstream projects then build all projects leading to them to support testing
+    // If the package is a core package, azure-identity or arm-resources then build only the package,
+    // otherwise build the package and all its dependents
+    var rushCommandFlag = "--impacted-by";
+
+    if (restrictedToPackages.includes(packageName)) {
+      // if this is one of our restricted packages with a ton of deps, make it targeted
+      // as including all dependents will be too much
+      rushCommandFlag = "--to";
+    } else if (actionComponents.length == 1) {
+      // else we are building the project and its dependents
+      rushCommandFlag = "--from";
+    }
+
+    mappedPackages.push([rushCommandFlag, packageName]);
+  }
+
+  return mappedPackages;
+};
+
+/**
+ * Returns an array of full paths to package.json files under a directory
+ *
+ * @param {string} searchDir - directory to search
+ */
+const getPackageJSONs = (searchDir) => {
+  // This gets all the directories with package.json at the `sdk/<service>/<service-sdk>` level excluding "arm-" packages
+  const sdkDirectories = fs
+    .readdirSync(searchDir)
+    .map((f) => path.join(searchDir, f, "package.json")); // turn potential directory names into package.json paths
+
+  // This gets all the directories with package.json at the `sdk/<service>/<service-sdk>/perf-tests` level excluding "-track-1" perf test packages
+  let perfTestDirectories = [];
+  const searchPerfTestDir = path.join(searchDir, "perf-tests");
+  if (fs.existsSync(searchPerfTestDir)) {
+    perfTestDirectories = fs
+      .readdirSync(searchPerfTestDir)
+      .filter((f) => !f.endsWith("-track-1")) // exclude libraries ending with "-track-1" (perf test projects)
+      .map((f) => path.join(searchPerfTestDir, f, "package.json")); // turn potential directory names into package.json paths
+  }
+
+  return sdkDirectories.concat(perfTestDirectories).filter((f) => fs.existsSync(f)); // only keep paths for files that actually exist
+};
+
+/**
+ * Returns package names and package dirs arrays
+ *
+ * @param {string} baseDir -
+ * @param {string[]} serviceDirs -
+ * @param {string} artifactNames -
+ */
+export const getServicePackages = (baseDir, serviceDirs, artifactNames) => {
+  const packageNames = [];
+  const packageDirs = [];
+  let validSdkTypes = ["client", "mgmt", "perf-test", "utility"]; // valid "sdk-type"s that we are looking for, to be able to apply rush-runner jobs on
+  console.log(`Packages to build: ${artifactNames}`);
```

Comment:
should we move these console.logs outside of this helper? it would feel weird for tests to log here

Reply:
good point!